internal/parser/scanner/scanner.go

```diff
@@ -228,23 +192,43 @@ func (s *scanner) peekString(str string) bool {
 }
 
 // createToken creates a token with given parameters
-func (s *scanner) createToken(t token.Type) token.Token {
-	tk := token.New(s.line, s.col, s.start, s.pos-s.start, t, string(s.input[s.start:s.pos]))
+func (s *scanner) createToken(t token.Type, value string) token.Token {
+	tk := token.New(s.line, s.col, s.start, s.pos-s.start, t, value)
 	s.start = s.pos
 	return tk
 }
 
-// seekNextPos returns the position of the end of a keyword.
+// seekTokenEnd returns the position of the end of a keyword.
 // It takes the start position of the keyword.
-func (s *scanner) seekNextPos(start int) int {
-	for start < len(s.input) && !whiteSpace.Matches(s.input[start]) {
+func (s *scanner) seekTokenEnd(start int) int {
+	for start < len(s.input) && !unicode.IsSpace(s.input[start]) { //!whiteSpace.Matches(s.input[start]) {
 		start++
 	}
 	return start
 }
 
+// unexpectedRune is returned when the combination/sequence of a set of
+// runes is not expected. The value argument contains all the valid
+// sequence of runes encountered till now and is consumed on execution
+// of this function.
 func (s *scanner) unexpectedRune(value string) token.Token {
 	tk := token.New(s.line, s.col, s.start, s.pos, token.Error, value)
 	s.start = s.pos
+	s.acceptString(value)
 	return tk
 }
+
+// peekNextRune returns the next rune in order in the input.
+// It checks for the end of input and existance of space(s)
+// as the next rune. A rune is returned successfully only if
+// there is no whitespace or end of input encountered. In other
+// cases an error is returned.
+func (s *scanner) peekNextRune() (rune, error) {
+	if s.start < len(s.input)-1 {
+		r := s.input[s.start+1]
+		if !unicode.IsSpace(r) {
+			return r, nil
+		}
+	}
+	return '\\', errors.New("No desired input found")
```

Comment:
Return `0` instead of `\`. Also, document that this is returned.

Also,  what happens when there is a space? then an error is returned, right?

Also, isn't start the starting position of the token? shouldn't you use `pos` here?

Reply:
Okay I will return `0`. Its actually returning an empty rune rn. Apparently I should escape it and `\\` is the escape sequence. 

Yes, when theres a space, an error is returned. 
The error is just an indication of whether I have a token that I'd be interested in, so makes sense here I suppose.

`start` is at `pos` at this time, so I think both works. Let me take a look tho.

Reply:
Changed to `s.pos`